src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+        if (publicKeyPath) {
+          publicKey = fs.readFileSync(publicKeyPath).toString();
+        }
+
+        const peerInfo: PeerInfo = await client.getPeerInfo(nodePath, currentNode, publicKey);
+
+        if (base64) {
+          pkLogger(peerInfo.toStringB64(), PKMessageType.SUCCESS);
+        } else {
+          pkLogger('Peer Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.publicKey, PKMessageType.SUCCESS);
+
+          pkLogger('Peer Connected Address:', PKMessageType.INFO);
+          pkLogger(peerInfo.connectedAddress?.toString() ?? '', PKMessageType.SUCCESS);
+
+          pkLogger('Relay Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.relayPublicKey ?? '', PKMessageType.SUCCESS);
+        }
+      }),
+    );
+}
+
+function makeListPeersCommand() {
+  return new commander.Command('list')
+    .description('list all connected peers')
+    .alias('ls')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+
+        const publicKeys = await client.listPeers(nodePath);
+        if (publicKeys === undefined || publicKeys.length == 0) {
+          pkLogger('no peers exist', PKMessageType.INFO);
+        } else {
+          publicKeys.forEach((publicKey: string, index: number) => {
+            pkLogger(`${index + 1}: ${publicKey}`, PKMessageType.INFO);
+          });
+        }
+      }),
+    );
+}
+
+function makePingPeerCommand() {
+  return new commander.Command('ping')
+    .description('ping a connected peer')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const publicKeyPath = options.publicKey;
+        // read in publicKey if it exists
+        const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+        const successful = await client.pingPeer(nodePath, publicKey);
+
+        if (successful) {
+          pkLogger('peer successfully pinged', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('ping timed out', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeFindPeerCommand() {
+  return new commander.Command('find')
+    .description('find a peer based on a public key')
```

Comment:
This depends on the DHT right? As in you are in stealth or active mode. And you are listening for peers. Right now there's no social lookup.

If it is in fact a social lookup. I think these types of commands must be separate from p2p commands.

As in there is keynode p2p, and then there's social p2p.

Social p2p should be "plugin-oriented". And we should clearly differentiate commands that work at the level of keynodes and the commands that are working at the level of social networks.

We may even prefix them with `social` commands, and then plugins to the different social networks. That code must be isolated from the main pk memory space. So a child process that has its own memory space and not allowed to access the parent memory (fork + exec) could be used.